build/Dockerfile

```diff
@@ -8,26 +18,27 @@ COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
+# NOTE: downstream this needs to be a tarball, not a live download
 RUN go mod download
 
 # Copy the go source
 COPY . .
 
-# compile workspace controller binaries
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build \
+# compile workspace controller binaries, then webhook binaries
+RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
+  CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build \
   -a -o _output/bin/devworkspace-controller \
   -gcflags all=-trimpath=/ \
   -asmflags all=-trimpath=/ \
-  main.go
-
-# Compile webhook binaries
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build \
+  main.go && \
+  CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build \
```

Comment:
Why combine steps here? This is a builder image so it has no impact on overall image size, and this makes the build harder to read.

Reply:
intent was to not have to define the same ARCH value twice... but doesn't matter either way, as long as it's arch aware.